### ISSUE TITLE
Add support for VictoriaMetrics metrics storage backend

### ISSUE THREAD

**Issue by valyala** [k6io/k6#2003]
## Feature Description
It would be great to add support for [VictoriaMetrics](https://github.com/VictoriaMetrics/VictoriaMetrics/) as metrics storage backends for k6.
VictoriaMetrics supports data ingestion via Influx line protocol (see [these docs](https://docs.victoriametrics.com/#how-to-send-data-from-influxdb-compatible-agents-such-as-telegraf)). It supports PromQL-based query language - [MetricsQL](https://docs.victoriametrics.com/MetricsQL.html), which is better suited for typical monitoring tasks and graph building than InfluxQL or Flux. See [this PromQL tutorial](https://valyala.medium.com/promql-tutorial-for-beginners-9ab455142085) for details. VictoriaMetrics is compatible with Prometheus datasource in Grafana - see [these docs](https://docs.victoriametrics.com/#grafana-setup).

**Comment by imiric** [comment]
Hi, it looks interesting, I wasn't familiar with the project.

If it supports the Influx line protocol, then it should be supported by k6 already, no?

We're trying to limit the amount of output backends k6 supports directly, in favor of generic protocols that are supported by more than one service, so it's unlikely support for this will be part of k6 core. That said, it could be easily used as an [output extension](https://k6.io/blog/k6-v0.31/#output-extensions).

**Comment by valyala** [comment]
> If it supports the Influx line protocol, then it should be supported by k6 already, no?

Yes - it should work OK on data ingestion side. As for data querying side, then the [output extension](https://k6.io/blog/k6-v0.31/#output-extensions) for VictoriaMetrics looks like a way to go.

**Comment by na--** [comment]
I see that VictoriaMetrics also [supports statsd metrics](https://docs.victoriametrics.com/#how-to-send-data-from-graphite-compatible-agents-such-as-statsd), which k6 is capable of outputting, and [supports Prometheus remote_write](https://docs.victoriametrics.com/#how-to-import-time-series-data), which we plan to add in k6 (https://github.com/k6io/k6/issues/1761). So given that we currently have 2 ways to send data to it, and we plan to have a 3rd one in the near future, I'll move this issue to the docs repo, given that I don't think we need a 4th way to send data to it, but should document the ones we already have... :sweat_smile:
